### PR TITLE
Add RTL support

### PR DESCRIPTION
**Pull request by mparvazi**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Adds a CSS file for fixing RTL issues.
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | -
| How to test?      | -
| Possible impacts? | -

### CODE REVIEW

**Comment by mparvazi** [comment]
The `rtl.scss` can be used for adding CSS rules for RTL.
This file will be converted to `theme.rtlfix` file after build.
The `theme.rtlfix` will be used by [StylesheetGenerator](https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Localization/RTL/StylesheetGenerator.php#L195) for creating RTL version of `theme.css` file.
Content of this file will be added to `theme.css` for creating `theme_rtl.css` file.

File: webpack/webpack.parts.js [hunk 1]
```diff
@@ -64,7 +64,7 @@ exports.extractScss = ({mode = 'production'}) => ({
   },
   plugins: [
     new MiniCssExtractPlugin({
-      filename: 'css/[name].css',
+      filename: ({ chunk: {name} }) => name === 'rtl' ? 'css/theme.rtlfix' : 'css/[name].css',
```
Comment (NeOMakinG) [inline webpack/webpack.parts.js:67]:
```suggestion
      filename: ({ chunk: {name} }) => name === 'rtl' ? 'css/theme.rtlfix' : 'css/[name].css',
```

Is the theme.rtlfix a real maintainable solution? What about a simple rtl folder with .scss files that we can generate, and include it if the rtl is activated? I feel like it's more maintainable, wdyt?

Comment (mparvazi) [reply webpack/webpack.parts.js:67]: I had a problem with rtl.css in public/css/rtl.css
StylsheetGenerator will create this file: rtl_rtl.css 
Screenshot 2022-01-03 at 09-43-26 Theme Logo Theme • sidebar

Comment (NeOMakinG) [reply webpack/webpack.parts.js:67]: I'm not sure to understand

**Comment by mparvazi** [comment]
@NeOMakinG 

Let's try without theme.rtlfix

**Adding rtl.scss:**
```
const entriesArray = {
  theme: ['scss', 'ts'],
  error: ['scss'],
  rtl: ['scss'],
};
```
**After build:**

**Now in BO we have to adopt theme-refacto for RTL:**
Design >  Theme & Logo >  Generate RTL stylesheet 



`rtl_rtl.css` will be added to pages.

**Comment by NeOMakinG** [comment]
> @NeOMakinG
> 
> Let's try without theme.rtlfix
> 
> **Adding rtl.scss:**
> 
> ```
> const entriesArray = {
>   theme: ['scss', 'ts'],
>   error: ['scss'],
>   rtl: ['scss'],
> };
> ```
> 
> **After build:**  **Now in BO we have to adapt theme-refacto for RTL:** Design > Theme & Logo > Generate RTL stylesheet
> 
> 
> 
> File `rtl_rtl.css` will be added to pages:
> 
> ```
> <link rel="stylesheet" href="http://localhost/develop/themes/theme-refacto/assets/css/rtl_rtl.css" type="text/css" media="all">
> ```

I think we don't even need the generate RTL mechanic in the new theme honestly, we can directly include the rtl.css stylesheet if rtl is enabled right?

**Comment by mparvazi** [comment]
You're right!

Problem:
How the RTL version of  `theme.css` and `error.css`  should be generated?

**Comment by mparvazi** [comment]
## RTL support for theme-refacto

**How RTL support for FO works in PS (classic theme or third-party themes that doesn't implement RTL)**
First of all user have to adapt theme for RTL: `BO > Theme & Logo >  Generate RTL stylesheet: Yes`
Then `StylesheetGenerator` converts css files to RTL version by `CSSJanus` library. For each css file (e.g. theme.css) a _rtl.css file will be generated. If FO is RTL then `FrontController` automatically adds _rtl.css files to the page (e.g. theme_rtl.css)

## Method 1
Adding rtl.scss by webpack in src folder:
```
const entriesArray = {
  rtl: ['scss'],
};
```
Then rtl.css will be created in assets

Problem:
When user try to adapt theme-refactor for RTL (theme.css and error.css), `StylesheetGenerator` will create rtl_rtl.css file.
This file is LTR version of rtl.css and will be used by `FrontController` in FO :neutral_face: 


## Method 2
Using theme.rtlfix trick (this solution used by BO theme for RTL)
In "Method 1" we added rtl.scss but we have to convert this file to theme.rtlfix:
`filename: ({ chunk: {name} }) => name === 'rtl' ? 'css/theme.rtlfix' : 'css/[name].css',`
So in this case we will had these files in assets:
`theme.css / error.css / theme.rtlfix`
Then `StylesheetGenerator` converts css files to RTL version but append content of theme.rtlfix without any changes to theme_rtl.css: `theme_rtl.css = theme.css + theme.rtlfix`

Problem:
@NeOMakinG doesn't like this method :smile: 

## Method 3
RTL support will be implemented inside theme-refacto.
**Step 1:** Creating RTL files with cssjanus plugin:
```
const cssjanus = require('cssjanus');

for (entry in css_files) {
    //Skip if filename is rtl.css
    //Convert file content to RTL: cssjanus.transform(entry.source());
    //Create RTL file
}
```
So we will had these files in assets:
`theme.css / theme_rtl.css / error.css / error_rtl.css / rtl.css`

In `theme.yml` we have to add an option like this:
`rtl_support: yes`

**Step 2:** Some minor changes in BO 
Let's back to this setting:
`BO > Theme & Logo >  Generate RTL stylesheet`
All themes will be displayed in dropdown menu for selecting by user.
But if theme has `rtl_support: yes` in its config, then it shouldn't be inside list items, or after selecting by user `Generate RTL stylesheet`should be disabled.

classic `rtl_support: no`
Screenshot 2022-01-06 at 08-28-21 Theme Logo Theme • develop

theme-refacto `rtl_support: yes`
Screenshot 2022-01-06 at 08-27-46 Theme Logo Theme • develop

So `FrontController` will be used _rtl.css files AND rtl.css automatically, but there is no problem because we don't have rtl_rtl.css file!

## Method 4
We use "Method 1" first:
assets: `theme.css / error.css / rtl.css`

Then adding a condition to `StylesheetGenerator` for using CSSJanus:
```
If (filename !== 'rtl.css')  {
    CSSJanus::transform($content);
}
```
So we wont have rtl_rtl.css file:
assets: `theme.css / theme_rtl.css / error.css / error_rtl.css / rtl.css`

As like as "Method 3" `FrontController` will be used _rtl.css files AND rtl.css automatically.

**Comment by NeOMakinG** [comment]
For clarification: It's not that I don't like the Method 2, but this way we don't benefit about SCSS and bootstrap variables/mixins and linters...

**Comment by NeOMakinG** [comment]
Method 3 looks the best to me because it permit to disable the RTL generation in the BO on the theme side, so if you want to support it yourself you can

**Comment by NeOMakinG** [comment]
While method 4 is not strict enough to me because if the theme use another name than rtl.css, it will be fucked up

**Pull request closed without merging**